algosdk/template.py

```diff
@@ -175,6 +175,89 @@ def get_program(self):
+    for N Algos, swap for Rate * N Assets.
+    ...
+
+    Args:
+        owner (str): an address that can receive the asset after the expiry
+            round
+        asset_id (int): asset to be transfered
+        ratn (int): the numerator of the exchange rate
+        ratd (int): the denominator of the exchange rate
+        expiry_round (int): the round on which the assets can be transferred
+            back to owner
+        min_trade (int): the minimum amount (of Algos) to be traded away
+        max_fee (int): the maximum fee that can be paid to the network by the
+            account
+
+    """
+    def __init__(self, owner: str, asset_id: int, ratn: int, ratd: int,
+                 expiry_round: int, max_fee: int, min_trade: int):
+        self.owner = owner
+        self.ratn = ratn
+        self.ratd = ratd
+        self.expiry_round = expiry_round
+        self.min_trade = min_trade
+        self.max_fee = max_fee
+        self.asset_id = asset_id
+
+    def get_program(self):
+        """
+        Return a byte array to be used in LogicSig.
+        """
+        orig = ("ASAKAAEFAgYEBwgJHSYBIJKvkYTkEzwJf2arzJOxERsSogG9nQzKPkpIoc" +
+                "4TzPTFMRYiEjEQIxIQMQEkDhAyBCMSQABVMgQlEjEIIQQNEDEJMgMSEDMB" +
+                "ECEFEhAzAREhBhIQMwEUKBIQMwETMgMSEDMBEiEHHTUCNQExCCEIHTUENQ" +
+                "M0ATQDDUAAJDQBNAMSNAI0BA8QQAAWADEJKBIxAiEJDRAxBzIDEhAxCCIS" +
+                "EBA=")
+        orig = base64.b64decode(orig)
+        offsets = [5, 7, 9, 10, 11, 12, 16]
+        values = [self.max_fee, self.min_trade, self.asset_id, self.ratd,
+                  self.ratn, self.expiry_round, self.owner]
+        types = [int, int, int, int, int, int, "address"]
+        return inject(orig, offsets, values, types)
+
+    def get_swap_assets_transactions(self, amount: int, contract: bytes,
+                                     private_key: str, first_valid,
+                                     last_valid, gh, fee):
+        """
+        Return a group transactions array which transfer funds according to
+        the contract's ratio.
+
+        Args:
+            amount (int): the amount of assets to be sent
+            contract (bytes): the contract containg information, should be
+                recieved from payer
+            private_key (str): the secret key to sign the contract
+            first_valid (int): first valid round for the transactions
+            last_valid (int): last valid round for the transactions
+            gh (str): genesis hash in base64
+            fee (int): fee per byte
+        """
+        txn_1 = transaction.PaymentTxn(self.get_address(), fee,
+                                       first_valid, last_valid, gh,
+                                       account.address_from_private_key(
+                                       private_key), int(
+                                           amount/self.ratn*self.ratd))
```

Comment:
I'm not sure we should compute this for the buyer automatically. We could verify that the amount is correct, but computing it for them could lead to mistakes (i.e. if someone sends them a contract with the wrong ratn/ratd)